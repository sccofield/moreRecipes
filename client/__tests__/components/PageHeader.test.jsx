import React from 'react';
import toJSON from 'enzyme-to-json';

import { Header, mapDispatchToProps }
  from '../../src/components/PageHeader';

describe('Upvote component', () => {
  it('should render the component', () => {
    const wrapper = shallow(<Header />);
    expect(wrapper.find('header').exists).toBeTruthy();
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});

describe('conatiner functions', () => {
  test('mapDispatchToProps', () => {
    const dispatch = jest.fn();
    expect(mapDispatchToProps(dispatch))
      .toHaveProperty('logoutUser');

    const { logoutUser } = mapDispatchToProps(dispatch);
    logoutUser();
    expect(dispatch).toHaveBeenCalled();
  });
});
