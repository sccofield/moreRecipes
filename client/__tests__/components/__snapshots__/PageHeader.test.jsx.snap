// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Upvote component should render the component 1`] = `
<header
  style={
    Object {
      "backgroundColor": "black",
      "backgroundImage": "none",
    }
  }
>
  <div
    className="container"
  >
    <Connect(Navbar) />
  </div>
</header>
`;
